### PR TITLE
Fix mutable default arguments in http3_client example

### PR DESCRIPTION
**Pull request by johnthagen**
Fix usage of mutable default arguments.

For an explanation of why mutable default arguments are a bad practice, see https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments.

### CODE REVIEW

**Comment by jlaine** [comment]
Hi! This PR makes sense, but could you please fix the linter errors and code coverage drop?

**Comment by jlaine** [comment]
This PR makes sense, but it looks as though I need to fix a linter error unrelated to your changes.

**Comment by johnthagen** [comment]
@jlaine Sounds great. Feel free to merge `main` into this branch when you are done as needed, or ping me to do it.

File: examples/http3_client.py [hunk 1]
```diff
@@ -129,26 +136,39 @@ def __init__(self, *args, **kwargs) -> None:
...
         )
 
-    async def post(self, url: str, data: bytes, headers: Dict = {}) -> Deque[H3Event]:
+    async def post(
+        self, url: str, data: bytes, headers: Optional[Dict] = None
+    ) -> Deque[H3Event]:
         """
         Perform a POST request.
         """
+        if headers is None:
```
Comment (jlaine) [inline examples/http3_client.py:156]: This seems unnecessary if HttpRequest accepts None

File: examples/http3_client.py [hunk 2]
```diff
@@ -129,26 +136,39 @@ def __init__(self, *args, **kwargs) -> None:
...
         )
 
-    async def websocket(self, url: str, subprotocols: List[str] = []) -> WebSocket:
+    async def websocket(
+        self, url: str, subprotocols: Optional[List[str]] = None
+    ) -> WebSocket:
         """
         Open a WebSocket.
         """
+        if subprotocols is None:
```
Comment (jlaine) [inline examples/http3_client.py:169]: This doesn't seem necessary, as there is an `if subprotocols` later in this function

File: examples/http3_client.py (hunk 1 above)
Comment (johnthagen) [reply examples/http3_client.py]: Fixed.

File: examples/http3_client.py (hunk 2 above)
Comment (johnthagen) [reply examples/http3_client.py]: Fixed.

**Comment by jlaine** [comment]
Thanks for this @johnthagen , merged!

**Pull request merged**